package/src/manifest.rs

```diff
@@ -230,15 +298,35 @@ impl ManifestFile {
     }
 
     /// Regenerate the lock file, even if it already exists.
-    pub fn regenerate_lock(&self, config: Config) -> Result<(LockFile, Snapshot), Error> {
-        let snap = self.snapshot_dependencies(config)?;
-        let lock = LockFile::new(self, &snap)?;
-
-        Ok((lock, snap))
+    pub fn regenerate_lock(&self, config: Config) -> Result<(LockFile, Resolution), Error> {
+        let snap = self.snapshot_dependencies(&config)?;
+        let has_index_pkg = snap.all_index_deps().next().is_some();
+        let index = if has_index_pkg {
+            // TODO: we could load the existing index first and check whether there the snapshot
+            // references any unknown index packages. If not, we could avoid refreshing the index.
+            match PackageIndex::refreshed(config.clone()) {
+                Ok(i) => i,
+                Err(e) => {
+                    eprintln!("failed to refresh package index ({e}), trying with the old one");
+                    PackageIndex::shared(config.clone())?
+                }
+            }
+        } else {
+            // TODO: we could avoid instantiating the index (which triggers a download if it doesn't exist)
+            // if the dependency tree has no index packages.
+            PackageIndex::shared(config.clone())?
+        };
+        // TODO: maybe refresh the index, if the snapshot has index dependencies?
```

Comment:
Aren't we doing that above?

Reply:
Indeed, I guess I forgot to remove the TODO